addons/account_edi_ubl_cii/tests/test_ubl_cii.py

```diff
@@ -0,0 +1,96 @@
+            'standard_price': 80.0,
+        })
+
+    def test_import_product(self):
+        line_vals = [
+            {
+                'product_id': self.place_prdct.id,
+                'product_uom_id': self.uom_units.id,
+                'tax_ids': [self.company_data_2['default_tax_sale'].id]
+            }, {
+                'product_id': self.displace_prdct.id,
+                'product_uom_id': self.uom_units.id,
+                'tax_ids': [self.company_data_2['default_tax_sale'].id]
+            }, {
+                'product_id': self.displace_prdct.id,
+                'product_uom_id': self.uom_units.id,
+                'tax_ids': [self.company_data_2['default_tax_sale'].id]
+            }, {
+                'product_id': self.displace_prdct.id,
+                'product_uom_id': self.uom_dozens.id,
+                'tax_ids': [self.company_data_2['default_tax_sale'].id]
+            }
+        ]
+        company = self.company_data_2['company']
+        company.country_id = self.env['res.country'].search([('code', '=', 'FR')])
+        company.vat = 'FR23334175221'
+        company.email = 'company@site.ext'
+        company.phone = '+33499999999'
+        company.zip = '78440'
+
+        company.partner_id.ubl_cii_format = 'facturx'
+        company.partner_id.bank_ids = [Command.create({
+            'acc_number': '999999',
+            'partner_id': company.partner_id.id,
+            'acc_holder_name': 'The Chosen One'
+        })]
+
+        invoice = self.env['account.move'].create({
+            'company_id': company.id,
+            'partner_id': company.partner_id.id,
+            'move_type': 'out_invoice',
+            'journal_id': self.company_data_2['default_journal_sale'].id,
+            'invoice_line_ids': [Command.create(vals) for vals in line_vals],
+        })
+        invoice.action_post()
+
+        template = self.env.ref('account.email_template_edi_invoice', raise_if_not_found=False)
+        print_wiz = self.env['account.move.send'].create({
+            'move_ids': invoice.ids,
+            'mail_template_id': template.id
+        })
+        print_wiz.checkbox_download = False
+        print_wiz.checkbox_send_mail = False
+        print_wiz.checkbox_send_by_post = False
+        print_wiz.checkbox_ubl_cii_xml = True
+        print_wiz.action_send_and_print()
+
+        facturx_attachment = invoice.ubl_cii_xml_id
+        xml_tree = etree.fromstring(facturx_attachment.raw)
+
+        # Testing the case where a product on the invoice has a UoM with a different category than the one in the DB
+        wrong_uom_line = xml_tree.findall('./{*}SupplyChainTradeTransaction/{*}IncludedSupplyChainTradeLineItem')[1]
+        wrong_uom_line.find('./{*}SpecifiedLineTradeDelivery/{*}BilledQuantity').attrib['unitCode'] = 'HUR'
+
+        facturx_attachment.raw = etree.tostring(xml_tree)
+        new_invoice = invoice.journal_id._create_document_from_attachment(facturx_attachment.ids)
```

Comment:
@sold-odoo Sorry I didn't notice your test was in account_edi_ubl_cii. You should test it on `l10n_account_edi_ubl_cii_tests` instead :x